### ISSUE TITLE
Bug: key ordering configuration not working as expected

### ISSUE THREAD

**Issue by cabello** [adrienverge/yamllint#178]
Hi 👋 

From the docs I understand that the entries are sorted using Unicode and not Lexicographical order. 

However I would expect the following:

```yaml
# yamllint disable rule:key-ordering
inherit_from: .rubocop_todo.yml

inherit_gem:
  ws-style:
    - default.yml
# yamllint enable rule:key-ordering

Metrics/LineLength:
  Max: 1680

RSpec:
  Enabled: false
```

To not output the following:
```
  9:1       error    wrong ordering of key "Metrics/LineLength" in mapping  (key-ordering)
  12:1      error    wrong ordering of key "RSpec" in mapping  (key-ordering)
```

My use case is: this project requires these entries (`inherit_*`) to be at the top of the file but I would like to sort everything past that hence enabling the key-ordering.

Thanks for sharing your work, I appreciate it. 💜

**Comment by adrienverge** [comment]
Thanks for the bug report @cabello,

I think that disabling the rule only works if comments completely wrap the
object of which keys needs to be sorted. For example:
```yaml
# yamllint disable rule:key-ordering
object:
  A: 1
  C: 3
  B: 2
# yamllint enable rule:key-ordering
```

Any contribution to solve this issue is welcome (if it doesn't break
anything else and comes with tests).

Le lun. 6 mai 2019 à 16:14, Danilo Cabello <notifications@github.com> a
écrit :

**Comment by cabello** [comment]
Hi @adrienverge 

I am trying to address this bug, I am having a few challenges finding a way to pick a specific test to run, I got all of them running by looking at `.travis.yml`. I wrote this test and added to `test_key_ordering.py` and it fails as expected:

```py
def test_disable_enable_ordering_with_comments(self):
        conf = 'key-ordering: enable'
        self.check('---\n'
            '# yamllint disable rule:key-ordering\n'
            'background: false\n'
            '# yamllint enable rule:key-ordering\n'
            'alphabet: true\n', conf)
```

Any hints on how I can run only this test so I can improve my REPL and eventually make it to the fix?

Thanks

**Comment by adrienverge** [comment]
Hi @cabello,

This is standard Python `unittest`, but I agree it's not easy to find how to run a single test 😉

Use this to run only tests for `key-ordering`:

    python3 -m unittest tests.rules.test_key_ordering

And this to run only your new test function:

    python3 -m unittest tests.rules.test_key_ordering.KeyOrderingTestCase.test_disable_enable_ordering_with_comments

**Comment by ssbarnea** [comment]
Any chance to be able to enforce the key-ordering in a single place, somehow using something like below seams to have no effect at all:
```
# yamllint enable rule:key-ordering
foo:
  zzz: {}
  aaa: {}
# yamllint disable rule:key-ordering
```

Users that have files that cannot be sorted (like playbooks) is unable to make use of key-ordering feature  because it cannot be enabled only where needed.

Any workarounds?
